packages/components/src/FormGroup/FormGroup.tsx

```diff
@@ -0,0 +1,107 @@
+import clsx from "clsx";
+import React, { ElementType, ReactNode } from "react";
+import Text from "../Text";
+import styles from "./FormGroup.module.css";
+
+type FormGroupProps = {
+  /**
+   * The contents of the form group. We suggest a FormGroup.Title followed by
+   * interactive elements.
+   *
+   * Should be wrapped in a fragment to allow our styling to control the spacing
+   * between elements.
+   */
+  children: ReactNode;
+  /**
+   * Type of element the immediate wrapper around the contents should be.
+   * @default 'fieldset'
+   */
+  as?: ElementType;
+  /**
+   * Additional classnames passed in for styling.
+   */
+  className?: string;
+};
+
+type TitleProps = {
+  /**
+   * The title of the form group.
+   */
+  children: ReactNode;
+  /**
+   * Type of element the wrapper should be.
+   * @default 'legend'
+   */
+  as?: ElementType;
+  /**
+   * Additional classnames passed in for styling.
+   */
+  className?: string;
+};
+
+type ItemsProps = {
+  /**
+   * The content of the helper form elements in the form group.
+   */
+  children: ReactNode;
+  /**
+   * Type of element the immediate wrapper around the contents should be.
+   * @default 'div'
+   */
+  as?: ElementType;
+  /**
+   * Additional classnames passed in for styling.
+   */
+  className?: string;
+};
+
+/**
+ * Helper sub-component for styling the title at the top of the component.
+ */
+const Title = ({
+  children,
+  as: Component = "legend",
+  className,
+}: TitleProps) => {
+  return (
+    <Component className={clsx(className, styles.title)}>
+      <Text size="sm" weight="bold">
+        {children}
+      </Text>
+    </Component>
```

Comment:
Does this result in HTML like with an extra P tag:

```html
<legend>
  <p class="stuff">Some text</p>
</legend>
```

?

If so, I wonder if it's worth thinking about reducing unnecessary elements. Maybe with something like

```jsx
<Text as={Component} className={clsx(className, styles.title)} size="sm" weight="bold">
  {children}
</Text>
```

Although that may not really be worth it. And would probably require updating the Text component to accept more things for `as`.

Reply:
https://app.shortcut.com/czi-edu/story/182120/allow-text-component-to-use-more-elements-for-as-prop